src/bin/utils.ts

```diff
@@ -196,6 +207,35 @@ async function requestPassword(): Promise<string> {
 //   return;
 // }
 
+const envVariables = {
+  PK_PASSWORD: process.env.PK_PASSWORD,
+  PK_RECOVERY_CODE: process.env.PK_RECOVERY_CODE,
+}
+
+async function getByFile(passwordFile: string | undefined): Promise<string | undefined> {
+  if (passwordFile == null) return;
+  // Reading the file.
+  const contents = await fs.promises.readFile(passwordFile, {
+    encoding: 'utf-8',
+  });
+  return contents.trim();
+}
+
+async function getPassword(passwordFile: string | undefined) {
+  // we have 3 places to get the password.
+  // 1. passed via CLI --password-file
+  // 2. passed via ENV PK_PASSWORD
+  // 3. passed via input
+  return (await getByFile(passwordFile)) ?? envVariables.PK_PASSWORD ?? await requestPassword();
+}
+
+async function getRecoveryCode(recoverCodeFile: string | undefined) {
+  // we have 2 places to get the recoveryCode.
+  // 1. passed via CLI --recovery-code-file
+  // 2. passed via ENV PK_RECOVERY_CODE
+  return (await getByFile(recoverCodeFile)) ?? envVariables.PK_RECOVERY_CODE;
+}
+
```

Comment:
Can some return types be added to these? My understanding is that `getPassword` should be `Promise<string>`, but `getRecoveryCode` will be `Promise<string | undefined>`.

Reply:
Added return types.